pkg/metrics/percentile/gk_array.go

```diff
@@ -253,15 +331,20 @@ func (s GKArray) Merge(o GKArray) GKArray {
 func (s GKArray) compressWithIncoming(incomingEntries Entries) GKArray {
 	// TODO[Charles]: use s.Incoming and incomingEntries directly instead of merging them prior to compressing
 	if len(s.Incoming) > 0 {
-		newIncoming := make([]Entry, len(incomingEntries), len(incomingEntries)+len(s.Incoming))
-		copy(newIncoming, incomingEntries)
-		incomingEntries = newIncoming
+		incomingCopy := make([]Entry, len(incomingEntries), len(incomingEntries)+len(s.Incoming))
+		copy(incomingCopy, incomingEntries)
+		incomingEntries = incomingCopy
 		for _, v := range s.Incoming {
 			incomingEntries = append(incomingEntries, Entry{V: v, G: 1, Delta: 0})
 		}
 	}
 	sort.Sort(incomingEntries)
 
+	// Copy Entries slice so as not to change the original
+	entriesCopy := make([]Entry, len(s.Entries), len(s.Entries))
+	copy(entriesCopy, s.Entries)
+	s.Entries = entriesCopy
```

Comment:
Could you add tests that trigger an error when not copying?

Probably better to be safe and copy for now 👍 . In the future we may want to look into cases where we are ok with modifying the underlying slices.

Reply:
Yes, added a test.